src/meta-srv/src/procedure/region_failover/failover_start.rs

```diff
@@ -95,6 +128,15 @@ impl State for RegionFailoverStart {
         ctx: &RegionFailoverContext,
         failed_region: &RegionIdent,
     ) -> Result<Box<dyn State>> {
+        let exist = self.failed_region_exist(ctx, failed_region).await?;
```

Comment:
Shouldn't this check have been placed in `RegionFailoverManager::do_region_failover`? The "wrong" failover procedure should not start at all.

Reply:
Before the procedure acquires the lock, the previous procedures may update the metadata. It needs to double-check it.